cli/connect/connect.go

```diff
@@ -0,0 +1,113 @@
+package connect
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/tarantool/cartridge-cli/cli/common"
+	"github.com/tarantool/cartridge-cli/cli/context"
+	"github.com/tarantool/cartridge-cli/cli/project"
+	"github.com/tarantool/cartridge-cli/cli/running"
+)
+
+const (
+	// see https://github.com/tarantool/tarantool/blob/b53cb2aeceedc39f356ceca30bd0087ee8de7c16/src/box/lua/console.c#L265
+	tarantoolWordSeparators = "\t\r\n !\"#$%&'()*+,-/;<=>?@[\\]^`{|}~"
+)
+
+func Enter(ctx *context.Ctx, args []string) error {
+	var err error
+
+	if err := FillCtx(ctx, args); err != nil {
+		return err
+	}
+
+	if ctx.Running.Instances, err = common.GetInstancesFromArgs(args, ctx); err != nil {
+		return err
+	}
+
+	if len(ctx.Running.Instances) != 1 {
+		return fmt.Errorf("Should be specified one instance name")
+	}
+
+	instanceName := ctx.Running.Instances[0]
+
+	process := running.NewInstanceProcess(ctx, instanceName)
+	if !process.IsRunning() {
+		return fmt.Errorf("Instance %s is not running", instanceName)
+	}
+
+	socketPath := project.GetInstanceConsoleSock(ctx, instanceName)
+	title := project.GetInstanceID(ctx, instanceName)
+
+	connOpts := ConnOpts{
+		Network: "unix",
+		Address: socketPath,
+	}
+
+	if err := runConsole(&connOpts, title); err != nil {
+		return fmt.Errorf("Failed to run interactive console: %s", err)
+	}
+
+	return nil
+}
+
+func Connect(ctx *context.Ctx, args []string) error {
+	if len(args) != 1 {
+		return fmt.Errorf("Should be specified one connection string")
+	}
+
+	connString := args[0]
+
+	connOpts, err := getConnOpts(connString, ctx)
+	if err != nil {
+		return fmt.Errorf("Failed to get connection opts: %s", err)
+	}
+
+	if err := runConsole(connOpts, ""); err != nil {
+		return fmt.Errorf("Failed to run interactive console: %s", err)
+	}
+
+	return nil
+
+}
+
+func FillCtx(ctx *context.Ctx, args []string) error {
+	var err error
+
+	if err := project.SetLocalRunningPaths(ctx); err != nil {
+		return err
+	}
+
+	if ctx.Running.AppDir == "" {
+		ctx.Running.AppDir, err = os.Getwd()
+		if err != nil {
+			return fmt.Errorf("Failed to get current directory: %s", err)
+		}
+	}
+
+	if ctx.Project.Name == "" {
+		if ctx.Project.Name, err = project.DetectName(ctx.Running.AppDir); err != nil {
+			return fmt.Errorf(
+				"Failed to detect application name: %s. Please pass it explicitly via --name",
+				err,
+			)
+		}
+	}
```

Comment:
There are at least three places with such code fragment. Consider some refactoring of FillCtx. Seems you need some GenericFillCtx to perform such checks.

Maybe not now but in this case please file an issue.

Reply:
See #427